src/renderer/viz/expressions/CategoryIndex.js

```diff
@@ -0,0 +1,142 @@
+import BaseExpression from './base';
+import { checkExpression, implicitCast, checkType, checkMaxArguments } from './utils';
+
+const SQRT_MAX_CATEGORIES_PER_PROPERTY = 256;
```

Comment:
We should document this limit in the expression docs. "The dataset must contain less than 65537 different categories"